File: 128-longest-consecutive-sequence/longest-consecutive-sequence.scala
object Solution {
    import scala.collection.mutable
    def longestConsecutive(nums: Array[Int]): Int = {
      val set: mutable.Set[Int] = collection.mutable.Set[Int]()
      nums.foreach(set.add)
      var longestStreak = 0
      for (num<- set)
        if (!set.contains(num - 1)) {
          var currentNum = num
          var currentStreak = 1
          while (set.contains(currentNum + 1)) {
            currentNum += 1
            currentStreak += 1
          }
          longestStreak = math.max(longestStreak, currentStreak)
        }
      longestStreak
    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.go
func longestConsecutive(nums []int) int {
	set := map[int]bool{}
	for _, num := range nums {
		set[num] = true
	}

	res := 0
	for _, num := range nums {
		if set[num-1] {
			continue
		}

		sequence, temp := 1, num + 1
		for set[temp] {
			sequence++
			temp++
		}

		res = max(res, sequence)
        if sequence > len(nums) / 2{
            break
        }
	}
	return res
}